### PR TITLE
refresher_spec: Refactor counts assertions, check volumes after deletions

### PR DESCRIPTION
**Pull request by cben**
Refactoring similar to https://github.com/ManageIQ/manageiq-providers-openshift/pull/75.
- compare together to get all differences printed together
- express counts after deletions as subtractions

Also added some checks for volumes after deletions, acounted for my-pod-2 also being deleted, removed some redundant lines.

(VCR coming soon, extracted this for easier review.)

@yaacov @enoodle @moolitayer Please review

### CODE REVIEW

File: spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb [hunk 1]
```diff
@@ -518,26 +535,24 @@ def assert_specific_persistent_volume_claim
...
-        [container0, container1].each do |container|
-          assert_disconnected(container)
+        [container0, container1, container2].each do |container|
           expect(container).not_to be_nil
-        end
-
-        container0 = Container.find_by(:name => "my-container")
-        container1 = Container.find_by(:name => "my-container")
-
-        [container0, container1].each do |container|
```
Comment (moolitayer) [inline spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb]: facepalm

**Comment by cben** [comment]
@zeari @yaacov Please review

**Comment by cben** [comment]
@moolitayer Can this be merged?  I'm building a followup on top of it...

**Comment by cben** [comment]
@miq-bot add-label gaprindashvili/yes
Needed for backporting later quotas test PRs, at least #198 
https://bugzilla.redhat.com/show_bug.cgi?id=1504560

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 33cb4d2fdced67872f5f61a0ca591cc5f295544f
Author: Mooli Tayer <mtayer@redhat.com>
Date:   Thu Jan 11 15:47:13 2018 +0200

    Merge pull request #193 from cben/vcr-spec-counts
    
    refresher_spec: Refactor counts assertions, check volumes after deletions
    (cherry picked from commit a80fca529c956c732afdc594402fdb8638192801)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1559544
```

**Pull request merged**